packages/xrpl/test/models/depositPreauth.test.ts

```diff
@@ -108,4 +157,154 @@ describe('DepositPreauth', function () {
       "DepositPreauth: Account can't unauthorize its own address",
     )
   })
+
+  it('throws when AuthorizeCredentials is not an array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list must be an array'
+    depositPreauth.AuthorizeCredentials = validCredential
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when UnauthorizeCredentials is not an array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list must be an array'
+    depositPreauth.UnauthorizeCredentials = validCredential
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when AuthorizeCredentials is empty array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list cannot be empty'
+    depositPreauth.AuthorizeCredentials = []
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when UnauthorizeCredentials is empty array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list cannot be empty'
+    depositPreauth.UnauthorizeCredentials = []
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when AuthorizeCredentials is too long', function () {
+    const sampleCredentials: AuthorizeCredential[] = []
+    const errorMessage =
+      'DepositPreauth: Credentials list cannot have more than 8 elements'
+    for (let index = 0; index < 9; index++) {
+      sampleCredentials.push({
+        Credential: {
+          Issuer: `SampleIssuer${index}`,
+          CredentialType: stringToHex('Passport'),
+        },
+      })
+    }
+    depositPreauth.AuthorizeCredentials = sampleCredentials
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when UnauthorizeCredentials is too long', function () {
+    const sampleCredentials: AuthorizeCredential[] = []
+    const errorMessage =
+      'DepositPreauth: Credentials list cannot have more than 8 elements'
```

Comment:
The word list can be excluded for array fields.
```suggestion
      'DepositPreauth: Credentials length cannot exceed 8 elements'
```